nuxeo-runtime/nuxeo-runtime-mongodb/src/main/java/org/nuxeo/runtime/mongodb/MongoDBConnectionHelper.java

```diff
@@ -82,17 +83,30 @@ public static MongoClient newMongoClient(String server) {
      * @since 10.3
      */
     public static MongoClient newMongoClient(MongoDBConnectionConfig config) {
+        return newMongoClient(config, null);
+    }
+
+    /**
+     * Initializes a connection to the MongoDB server.
+     *
+     * @param config the MongoDB connection config
+     * @param optionsConsumer a consumer of the client options builder
+     * @return the MongoDB client
+     * @since 11.1
+     */
+    public static MongoClient newMongoClient(MongoDBConnectionConfig config,
+                                             Consumer<MongoClientOptions.Builder> optionsConsumer) {
         String server = config.server;
         if (StringUtils.isBlank(server)) {
             throw new RuntimeException("Missing <server> in MongoDB descriptor");
         }
-        MongoClientOptions.Builder optionsBuilder = MongoClientOptions.builder()
-                  // can help to prevent firewall disconnecting inactive connection, option not available from URI
-                  .socketKeepAlive(true)
```

Comment:
Is it expected to remove `socketKeepAlive` ?

Reply:
Yes, the method is deprecated. The builder now defaults to true and disabling it is not recommended.

Reply:
Having said that, I will backport this fix, so perhaps I don't need to do this as part of this PR.  Do you know if we upgrade the mongo client regularly for 9.10 and 10.10?

Reply:
I know we upgraded it during the 10.10 (from 3.4.x to 3.8.y) within https://jira.nuxeo.com/browse/NXP-25620 and it seems that the method wasn't deprecated yet in 3.4, we're gonna need it for 10.10 and 9.10.

Reply:
Ok, thanks for confirming.  I will put it back when I do the backport PRs.